### PR TITLE
Feature/module import must throw exception

### PR DESCRIPTION
**Pull request by dizzzz**
Finally i found the issue: a missing line in a XPathException constructor; now a specific ErrorCode can be used to detect the situation.

### CODE REVIEW

File: src/org/exist/xquery/XPathException.java [hunk 1]
```diff
@@ -322,8 +362,39 @@ public String getMessageAsHTML() {
...
+     * 
+     * @return Error value as sequence
+     */
+    public Sequence getErrorVal() {
+        return errorVal;
+    }
+
     public static class FunctionStackElement {
 
+        String function;
```
Comment (adamretter) [inline src/org/exist/xquery/XPathException.java]: These should be final I think?

File: src/org/exist/xquery/XPathException.java [hunk 2]
```diff
@@ -322,8 +362,39 @@ public String getMessageAsHTML() {
...
+        String file;
+        int line;
+        int column;
+
+        public FunctionStackElement(UserDefinedFunction func, String file, int line, int column) {
```
Comment (adamretter) [inline src/org/exist/xquery/XPathException.java]: Likewise, I think these should be final...

File: src/org/exist/xquery/functions/inspect/ModuleFunctions.java [hunk 3]
```diff
@@ -60,15 +60,26 @@ public Sequence eval(Sequence[] args, Sequence contextSequence) throws XPathExce
...
+                if (isCalledAs("module-functions-by-uri")) {
+                    module = tempContext.importModule(args[0].getStringValue(), null, null);
+                } else {
+                    module = tempContext.importModule(null, null, args[0].getStringValue());
+                }
+                
+            } catch (final XPathException e) {
+                LOG.debug("Failed to import module: " + args[0].getStringValue() + ": " + e.getMessage(), e);
+
+                if (e.getErrorCode().equals(ErrorCodes.XPST0003)) {
```
Comment (adamretter) [inline src/org/exist/xquery/functions/inspect/ModuleFunctions.java]: I don't think XPST0003 is the correct error message to use here? It is for static error, but this is at runtime so it must by a dynamic error...

Comment (adamretter) [inline src/org/exist/xquery/functions/inspect/ModuleFunctions.java]: Also, why would you hide all other XPathExceptions from the user and send them only to the log?

Comment (dizzzz) [inline src/org/exist/xquery/functions/inspect/ModuleFunctions.java]: about also: that was the current design; all was swallowed, agreed with wolfgang to do so.....

Comment (dizzzz) [inline src/org/exist/xquery/functions/inspect/ModuleFunctions.java]: about XPST0003: we hava a syntactic error in a module, this code looked best matching to me

File: src/org/exist/xquery/XPathException.java (hunk 2 above)
Comment (dizzzz) [inline src/org/exist/xquery/XPathException.java]: hmmm yes, my IDE did not show it though :-(

File: src/org/exist/xquery/XPathException.java [hunk 4]
```diff
@@ -322,8 +362,39 @@ public String getMessageAsHTML() {
...
+     * 
+     * @return Error value as sequence
+     */
+    public Sequence getErrorVal() {
+        return errorVal;
+    }
+
     public static class FunctionStackElement {
 
+        private final String function;
```
Comment (dizzzz) [inline src/org/exist/xquery/XPathException.java]: made final

**Comment by adamretter** [comment]
Okay cool, let me run it through the test suite here and then I am happy.

**Comment by dizzzz** [comment]
:)

**Comment by adamretter** [comment]
All passes here :-)

**Pull request merged**